config/initializers/line_item_controller_gift_card_details.rb

```diff
@@ -0,0 +1,3 @@
+Rails.application.config.to_prepare do
+  Spree::Api::LineItemsController.line_item_options += [:gift_card_details]
```

Comment:
do we also want to take into account the case gets created through another route? e.g. a nested attribute of order creation? maybe updating permitted_attributes rather than just in this controller?